core/src/main/java/com/linecorp/armeria/client/retry/RetryStrategy.java

```diff
@@ -34,53 +37,90 @@
 @FunctionalInterface
 public interface RetryStrategy<I extends Request, O extends Response> {
 
+    Backoff defaultBackoff = Backoff.of(Flags.defaultBackoffSpec());
+
     /**
      * A {@link RetryStrategy} that defines a retry should not be performed.
      */
     static <I extends Request, O extends Response> RetryStrategy<I, O> never() {
         return new RetryStrategy<I, O>() {
             @Override
-            public CompletableFuture<Boolean> shouldRetry(I request, O response) {
-                return CompletableFuture.completedFuture(false);
+            public CompletableFuture<Optional<Backoff>> shouldRetry(I request, O response) {
+                return CompletableFuture.completedFuture(Optional.empty());
             }
 
             @Override
-            public boolean shouldRetry(I request, Throwable throwable) {
-                return false;
+            public Optional<Backoff> shouldRetry(I request, Throwable throwable) {
+                return Optional.empty();
             }
         };
     }
 
     /**
-     * Returns the {@link RetryStrategy} that decides to retry the request using HTTP statuses.
+     * Returns the {@link RetryStrategy} that decides to retry the request using HTTP statuses with the
+     * {@link #defaultBackoff}.
+     */
+    static RetryStrategy<HttpRequest, HttpResponse> onStatus(HttpStatus retryStatus) {
+        return onStatus(retryStatus, defaultBackoff);
+    }
+
+    /**
+     * Returns the {@link RetryStrategy} that decides to retry the request using HTTP statuses with the
+     * specified {@link Backoff}.
+     */
+    static RetryStrategy<HttpRequest, HttpResponse> onStatus(HttpStatus retryStatus, Backoff backoff) {
```

Comment:
This method is maybe, just too much. I will remove this and above one. Thank you~!